### DISCUSSION TITLE
Have Macos builds been dropped from SQLAclchemy 1.4, was this deliberate.

### DISCUSSION THREAD

**Discussion by couling** [sqlalchemy/sqlalchemy#10868] [Usage Questions] [answered]
I've run into a build error trying to install the latest versions of sqlalchemy 1.4 on macos (intel).  It seems up to 1.4.49, sqlalchemy was publishing wheels for macos, but 1.4.50 and 1.4.51 are not published as wheels despite the latest 2.x releases being published for macos as wheels.

Was this intentional and are they expected to return?

**Comment by couling** [comment]
Now I look at it, it seems the Github action failed to complete for the past two 1.4 releases. Eg: https://github.com/sqlalchemy/sqlalchemy/actions/runs/6685640029

**Comment by zzzeek** [comment]
edit: when poster referred to "failed builds" I assumed they were referring to the py2.7 builds.    the actual osx/windows actions are not even running

> 
> Was this intentional 

no, we would have liked the py2.7 builds to continue

> and are they expected to return?

github itself has dropped Python 2.7 availability from the actions platform that we rely upon to build the wheels, so at the moment there is no pathway to getting Py2.7 builds going.   you can always install from the source .tar.gz and just skip the cextensions build.

**Reply by zzzeek** [reply]
the 2.7 builds are the failing builds.  as far as the macosx platform and windows platform, yah those are missing

**Reply by CaselIT** [reply]
that's unclear why. I'll investigate

**Reply by couling** [reply]
It probably goes without saying, but just to avoid any doubt this is affecting me using Macos with Python 3.10 so not a "Python 2 issue" from my side.

Thanks for following up.

**Reply by couling** [reply]
And Sadly I can't install from source.  I came here because I ran into an error with `pip install sqlalchemy==1.4.51`:

```
        File "/usr/local/Cellar/python@3.10/3.10.12/Frameworks/Python.framework/Versions/3.10/lib/python3.10/functools.py", line 873, in register
          raise TypeError(
      TypeError: Invalid annotation for 'args'. (<class 'collections.abc.Mapping'>, (), ()) is not a class.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```
Do you want me to raise this as an issue ticket?

**Reply by zzzeek** [reply]
not yet...is there a stack trace for that?

**Comment by CaselIT** [comment]
> Was this intentional and are they expected to return?

it seems that no build is present for windows or linux because github has not been able to run the action.

I fail to see way that is the case, since looking at the yaml it seems correct...

**Reply by zzzeek** [reply]
yah I just noticed this.  this has to be fixed

**Reply by CaselIT** [reply]
me too

**Comment by CaselIT** [comment]
It's unclear to me what of the changes in https://github.com/sqlalchemy/sqlalchemy/compare/rel_1_4_49...rel_1_4_50 has made github fail with 
> Create wheel: .github#L1
> Error when evaluating 'runs-on' for job 'make-wheel-win-osx'. .github/workflows/create-wheels.yaml (Line: > 19, Col: 14): Unexpected value ''


I'll take a look..

**Reply by CaselIT** [reply]
oh maybe they removed the latest thing? at least I see that on 2.0 we don't have it anymore
let me try

**Reply by CaselIT** [reply]
no that does not seems it, since the where workflow uses the `latest` names without issues..

**Reply by CaselIT** [reply, marked as answer]
ok seems fixed now: https://github.com/CaselIT/sqlalchemy/actions/runs/7493430220 and https://pypi.org/project/SQLAlchemy/1.4.50/#files

will do also 51 and update 1_4 to have the fixed file

**Reply by zzzeek** [reply]
did you change something or they just werent working when they ran ?

**Reply by CaselIT** [reply]
this is the fix https://github.com/sqlalchemy/sqlalchemy/commit/0653c1ca11f4f5ac8ee17144d78afe71740a88e8
